digicampipe/image/sky_image.py

```diff
@@ -0,0 +1,555 @@
+                                    fill=False, color='Y')
+                    ax.add_artist(circle)
+                    ax.text(star_px[0] + crop_pixel1[0], star_px[1] + crop_pixel1[1], gamma_star_name, color='Y')
+                crab_nebula_px = sky_image.wcs.wcs_world2pix(83.640187, 22.044295, 1)
+                circle = Circle((crab_nebula_px[0] + crop_pixel1[0], crab_nebula_px[1] + crop_pixel1[1]), radius=20,
+                                fill=False, color='Y')
+                ax.add_artist(circle)
+                ax.text(crab_nebula_px[0] + crop_pixel1[0], crab_nebula_px[1] + crop_pixel1[1], 'Crab Pulsar', color='Y')
+            width = crop_pixel2[0] - crop_pixel1[0]
+            height = crop_pixel2[1] - crop_pixel1[1]
+            rect = Rectangle(crop_pixel1, width=width, height=height, fill=False, color='y', linestyle='dashdot')
+            ax.add_artist(rect)
+        # plt.imshow(image_treated, cmap='gray')
+        plt.plot(0, 0, 'w+')
+        plt.grid(None)
+        plt.axis('off')
+        ax.get_xaxis().set_visible(False)
+        ax.get_yaxis().set_visible(False)
+        if output_dir is None:
+            plt.show()
+        else:
+            output_filename = self.filename.replace('.fits', '-treated.png')
+            plt.savefig(output_filename, bbox_inches='tight', pad_inches=0)
+            plt.close(fig)
+            print(output_filename, 'saved.')
+
+
+class LidCCDObservation:
+    """
+    LidCCDObservation regroups several LidCCDImages. Useful to find moving stars.
+    """
+    def __init__(self, filenames, crop_pixels1, crop_pixels2, threshold=None,
+                 scale_low_images_deg=None, scale_high_images_deg=None,
+                 guess_ra_dec=None, guess_radius=None):
+        """
+        Create a lid ccd observation from several pictures of the lid CCD camera.
+        Pointing is determined for each of the regions defined.
+        :param filenames: list of strings containing the path to lid CCD images (fits files).
+        :param crop_pixels1: list of points defining the upper left corner of the region of interest in pixels
+        :param crop_pixels2: list of points defining the lower right corner of the region of interest in pixels
+        :param threshold: threshold: pixels with a smaller value than threshold after subtraction are set to 0. Default at 100x the
+            average pixel value after subtraction.
+        :param scale_low_images_deg: minimum field of view (in degrees) considered during fitting.
+        :param scale_high_images_deg: maximum field of view (in degrees) considered during fitting.
+        :param guess_ra_dec: aproximate pointing coordinates
+        :param guess_radius: radius of pointing  coordinates around guess_ra_dec tried during determination
+        """
+        self.lidccd_images = []
+        image_shape = None
+        if type(filenames) is not list or len(filenames) == 0:
```

Comment:
this check is not actually needed. two lines below this check in:

    for filename in filenames:
    
An exception will be raised, if filenames is an empty list.

It is not really advisable to explicitely check for the type `list` here. Since also a `tuple` of filenames would be okay. Also any iterable would be okay, which spits out filenames, when iterated over like `glob.iglob` for example.

Reply:
Explicite type checks are generally considered unpythonic... the mindset is: If it sounds like a duck and looks like a duck, it probably is a duck.
That means, if my code can work with the stuff it was given, then I am fine with it, if not, the code will raise an exception in exactly that place, where the problem occured. There is generally no need for me as a developer to explicitely check a type.